### PR TITLE
Implement additional basic setup

### PR DESCRIPTION
**Pull request by hayata-suenaga**
cc: @tgolen @thienlnam 

### Details
Implement additional setup for TypeScript and the build process

### Related Issues
GH_LINK https://github.com/Expensify/App/issues/22716

### Manual Tests
N/A

### Linked PRs
N/A

### CODE REVIEW

**Comment by hayata-suenaga** [comment]
I have read the CLA Document and I hereby sign the CLA

File: .prettierignore [hunk 1]
```diff
@@ -1,17 +1,17 @@
 # The GH actions don't seem to compile and verify themselves well when Prettier is applied to them
```
Comment (thienlnam) [inline .prettierignore]: These should be uncommented

**Review by thienlnam — CHANGES_REQUESTED** [review]
Looks good, just one comment

File: .prettierignore (hunk 1 above)
Comment (hayata-suenaga) [reply .prettierignore]: The `react-native-x-maps` library is working without these ignores. Removing the file `.prettierignore` all together.

**Comment by thienlnam** [comment]
Ah I don't think we should remove it entirely - having it means `npm run prettier` won't bother running on those files which is a good thing. It doesn't make a difference in the commited code because we don't commit it, but speeds up the command locally

**Review by tgolen — DISMISSED** [review]

**Review by tgolen — APPROVED** [review]

**Review by thienlnam — APPROVED** [review]

**Pull request merged**